### ISSUE TITLE
Hapi 12+ and yar bug fix

### ISSUE THREAD

**Issue by pabshazon** [blinkmobile/hapi-oauth2orize#13]
From https://github.com/hapijs/yar

> Starting with Hapi 12 the request.session placeholder was removed.
> The guidance from Hapi maintainer Eran Hammer was for this and similar modules to move data storage away from request.session and use a more unique location. So, starting in 6.x.x the yar storage has been moved to request.yar

My solution:

index.js line 201 should be:
`request.yar.lazy(true);`

index.js line 206 should be:
`session: request.yar,`

Maybe you prefer a different approach to solve the issue

**Comment by devinivy** [comment]
Thanks!  This module does require a little bit of maintenance right now to bring it up-to-date.

**Comment by pabshazon** [comment]
Come on, we can resurrect it :)

**Comment by devinivy** [comment]
I don't think it will take much work, aside from the fact that it's officially time to write some proper tests for this plugin!

**Comment by lutfor3737** [comment]
@devinivy advance thanks for test implementation of this plugin. HapiJS community will be happy to get oauth2.

**Comment by pabshazon** [comment]
@devinivy, I can take care of the testing. 
What testing libraries would you use? 
I like Jasmine + Frisby but they might not be the best to test chained calls... any library recommendation to dev the tests for the Oauth2 process?

**Comment by devinivy** [comment]
@pabshazon I would personally use lab!
